src/Domain/CoordinateMaps/TimeDependent/SphericalCompression.cpp

```diff
@@ -0,0 +1,357 @@
+void check_source_radius(const T& radius, const double min_radius,
+                         const double max_radius) {
+  check_radius<InteriorMap, T>(radius, min_radius, max_radius,
+                               "Source radius "s);
+}
+
+template <bool InteriorMap, typename T>
+void check_target_radius(const T& radius, const double min_radius,
+                         const double max_radius) {
+  check_radius<InteriorMap, T>(radius, min_radius, max_radius,
+                               "Target radius "s);
+}
+}  // namespace
+
+namespace domain::CoordinateMaps::TimeDependent {
+template <bool InteriorMap>
+SphericalCompression<InteriorMap>::SphericalCompression(
+    std::string function_of_time_name, const double min_radius,
+    const double max_radius, const std::array<double, 3>& center) noexcept
+    : f_of_t_name_(std::move(function_of_time_name)),
+      min_radius_(min_radius),
+      max_radius_(max_radius),
+      center_(center) {
+  ASSERT(max_radius - min_radius > 1.0e-10,
+         "max_radius must be greater than min_radius, but max_radius - "
+         "min_radius == "
+             << max_radius - min_radius);
+}
+
+template <bool InteriorMap>
+template <typename T>
+std::array<ResultType<T>, 3> SphericalCompression<InteriorMap>::operator()(
+    const std::array<T, 3>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  const std::array<ResultType<T>, 3> source_rad_position{
+      radial_position(source_coords, center_)};
+  const ResultType<T> source_radius{magnitude(source_rad_position)};
+  check_source_radius<InteriorMap>(source_radius, min_radius_, max_radius_);
+  std::array<ResultType<T>, 3> result{};
+  for (size_t i = 0; i < 3; ++i) {
+    gsl::at(result, i) = gsl::at(source_coords, i);
+    correct_mapped_coordinate_or_frame_velocity<InteriorMap>(
+        make_not_null(&gsl::at(result, i)), gsl::at(source_rad_position, i),
+        source_radius, lambda00_y00(f_of_t_name_, time, functions_of_time),
+        min_radius_, max_radius_);
+  }
+  return result;
+}
+
+template <bool InteriorMap>
+std::optional<std::array<double, 3>> SphericalCompression<InteriorMap>::inverse(
+    const std::array<double, 3>& target_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  const double lambda_y{lambda00_y00(f_of_t_name_, time, functions_of_time)};
+  if (UNLIKELY(min_radius_ - max_radius_ > lambda_y or
+               lambda_y > min_radius_)) {
+    return {};  // map not invertible
+  } else {
+    const std::array<double, 3> target_radial_position{
+        radial_position(target_coords, center_)};
+    const double target_radius{magnitude(target_radial_position)};
+    check_target_radius<InteriorMap>(target_radius, min_radius_ - lambda_y,
```

Comment:
Why doesn't this return a `bool` and then you return a `std::nullopt` because you couldn't invert the map? Doesn't the current design interfere with the idea that `if we can't find the inverse, that's okay?` which we need for a number of different things.